### PR TITLE
8602 refactor odk codes

### PR DESCRIPTION
**Pull request by DevneyHamilton**
Not sure about this style of PORO. Seems best to return form item id instead of item itself so errors happen mostly in parser, but we should talk about it.

### CODE REVIEW

File: spec/models/odk/code_mapper_spec.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+      code = "qing#{form_item.id}"
+      expected = form_item.id
+      actual = Odk::CodeMapper.new.item_id_for_code(code, form)
+      expect(expected).to eq actual
+    end
+
+    it "retrieves correct questioning when it receives old style q{question.id} code" do
+      other_form = create(:form, question_types: ["text", %w[text text]])
+      other_form.c[0].update_attribute(:question_id, form.c[0].question_id)
```
[the remark this answers is not in the document]
Comment (DevneyHamilton) [reply spec/models/odk/code_mapper_spec.rb]: Not sure what to do with this - I do want to skip validations here.

File: spec/models/odk/code_mapper_spec.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      form_item = form.c[0]
+      code = "q#{form_item.question.id}"
+      expected = form_item.id
+      actual = Odk::CodeMapper.new.item_id_for_code(code, form)
+      expect(expected).to eq actual
+    end
+
+    it "handles multilevel codes" do
+      multilevel_qing = form.c[2]
+      codes = multilevel_qing.subqings.map { |sq| Odk::SubqingDecorator.decorate(sq).odk_code }
```
Comment (smoyth) [inline spec/models/odk/code_mapper_spec.rb]: Should this one's `odk_code` responsibilities also go in CodeMapper?

File: app/models/odk/code_mapper.rb [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      "on#{node_id}"
+    end
+
+    def option_id_for_code(code)
+      node_id = remove_prefix_if_matches(code, "on")
+      if node_id
+        OptionNode.id_to_option_id(node_id)
+      else
+        # fallback by looking up other inputs as option ids
+        Option.where(id: code).pluck(:id).first
```
[the remark this answers is not in the document]
Comment (smoyth) [reply app/models/odk/code_mapper.rb]: Update: delete!

**Comment by smoyth** [comment]
One more thing for this PR -- can you remove or implement this:

```
        context "with earlier odk code style for questionings" do
          xit "should parse the response" do
          end
        end
```

in spec/models/odk/response_parser_spec.rb?

Thx!!!

**Pull request merged**